### PR TITLE
Bulk combat changes; Armor Divisors, bullet balance, mob health/armor, etc.

### PR DESCRIPTION
**Pull request by cdb-is-not-good**
oh my GOD holy FUCK jesus CHRIST. ports https://github.com/discordia-space/CEV-Eris/pull/7353

Hooooookay. BIG PR, LOTS of changes. Bullets should generally be more lethal and less anemic and armor should be more gooder.

post-pen is dead and replaced by woundiing mod which seems to play better with armor. 

Armor is generally more effective at lower calibers and less effective at higher calibers. 

Most common mobs have lost much of their armor(particularly roaches/spiders. Termites were hit as well but not as hard). 

Many common mobs(roach+spider+termite) do not get modifiers anymore, only their "elites"(emperor/empress, fuhrer/panzer, queen+king termite). 

12mm pistols are back down to 7 for autos 5 for revolvers and have naturally received a buff along wiith all the other ballistics. 

Jungle birds also got a bit of a nerf to their overall armor but not very significant compared to others.

Attempts to full implement wounding multipliers and remove post-pen(which did a bit of a poor job at what we wanted it for)

Wield Delay Factor raised from 0 to 2, this will result in a slightly higher amount of time to wield(and thus higher vig needed to instant wield) but my testing didn't....really show that it was super significant. you can still instant wield a nagant with 25 vig(starting for BS) and i suspect this may need to be tweaked later. 

Siignificant screwing with mob balance

- [x] Runtimes surrounding all damage applications. LOTTA divide by 0 errors and fails. Preliminary testing indicated that applying a clamp(0.1-10) fixed it but that's likely a bandaid.
- [ ] LOTS of testing needed and several thumbings through of every single source of AP/armor.
- [ ] reevaluate all mobs armor and penetration as well as health - many still have values that're probably far too high especially with how this new system(theoretically) works
- [x] give another once over to make sure that all post-pen code is nuked
- [x] take a pass at wounding mults to make them properly apply to the HPs now that we are gettiing rid of post-pen.

## About The Pull Request
<details>
<summary>
	About The Pull Request
</summary>
<hr>

	
<hr>
</details>

## Changelog
:cl:

/:cl:

### CODE REVIEW

File: code/game/machinery/vendor/misc_vendors.dm [hunk 1]
```diff
@@ -82,7 +82,7 @@
 				/obj/item/flame/lighter/zippo/capitalist = 500, // Item description, meant to be expensive
 				/obj/item/flame/lighter/zippo/royal = 50,
 				/obj/item/flame/lighter/zippo/gonzo = 50,
-				/obj/item/flame/lighter/zippo/bullet = 50,
+				/obj/item/flame/lighter/zippo/bullet = 12,
```
Comment (Trilbyspaceclone) [inline code/game/machinery/vendor/misc_vendors.dm:85]: heh

File: code/game/objects/items/stacks/matter_synth.dm [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /datum/matter_synth
 	var/name = "Generic Synthesizer"
-	var/max_energy = 60000
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/stacks/matter_synth.dm:3]: ?

File: code/modules/clothing/spacesuits/rig/suits/light.dm [hunk 3]
```diff
@@ -44,13 +44,15 @@
 	icon_state = "hacker_rig"
 	suit_type = "SI 'Spacer"
 	armor_list = list(
-		melee = 10,
-		bullet = 5,
+		melee = 2,
+		bullet = 1,
 		energy = 0,
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/spacesuits/rig/suits/light.dm:49]:
```suggestion
		energy = 1,
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 4]
```diff
@@ -25,7 +25,7 @@
 		weapon_edge = 0
 
 	hit_impact(effective_force, get_step(user, src))
-	damage_through_armor(effective_force, I.damtype, hit_zone, ARMOR_MELEE, armour_pen = I.armor_penetration + (user.stats.getStat(STAT_ROB) * 0.25), used_weapon = I, sharp = weapon_sharp, edge = weapon_edge, post_pen_mult = I.post_penetration_dammult)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/carbon_defense.dm:28]: melee rob nerf V_V it should still scale a little imo

Comment (cdb-is-not-good) [reply code/modules/mob/living/carbon/carbon_defense.dm:28]: Probably, that got removed in the process of just changing over old->new code.

File: code/modules/clothing/spacesuits/rig/suits/light.dm (hunk 3 above)
Comment (cdb-is-not-good) [reply code/modules/clothing/spacesuits/rig/suits/light.dm:49]: wwwwhhhyyyyy?

File: code/game/objects/items/stacks/matter_synth.dm (hunk 2 above)
Comment (cdb-is-not-good) [reply code/game/objects/items/stacks/matter_synth.dm:3]: oh weird, dunno

File: code/game/machinery/vendor/misc_vendors.dm (hunk 1 above)
Comment (cdb-is-not-good) [inline code/game/machinery/vendor/misc_vendors.dm:85]:
```suggestion
				/obj/item/flame/lighter/zippo/bullet = 50,
```

File: code/game/objects/items/stacks/matter_synth.dm [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
+	var/max_energy = 15000
```
Comment (cdb-is-not-good) [inline code/game/objects/items/stacks/matter_synth.dm:3]:
```suggestion
	var/max_energy = 60000
```

**Comment by cdb-is-not-good** [comment]
merged for one round, unmerging.

**Comment by benj8560** [comment]
just gonna list my current problems I sent in DMs for reference:

Guild armor plating mods are currently untouched meaning they grant you 20 divisor (god tier armor)

Shotguns seem...odd all around? Slugs have 2 divisor thats like HV tier armor pen ontop of 45 damage and high wounding. Seems pretty nuts to me. Buckshots killing nightmare stalkers in 3-4 shots stats seem weird. Most of its damage is in its damage stat rather then wound scaling like I'd assume. Seems bad for PVE balance. Might be bigger mobs just need more armor, 

Hollowpoint can now be AP modded allowing it to reach alarming heights. I managed to get a 12mm to 1.7 AP on hollow point. I'm sure there's countless wacky combos that would allow for frankly ridiculous damage output especially on turbo AP guns like greyson weapons.  For PVE and more importantly PVP,

File: code/modules/clothing/spacesuits/rig/suits/light.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/modules/clothing/spacesuits/rig/suits/light.dm:49]: devide by 0?

**Comment by ChefDoggo** [comment]
image
no more attachment number for max y'can have, its so over

**Comment by benj8560** [comment]
Most of the prior problems I listed are solved aside from one: Simple mobs aka nightmares, renders, etc are still having some....odd behavors with guns but it seems _better_ at least. They do seem to take quiet a large amount of small arms fire before dying. 

New problems I've noticed listing from DMs once again for easy ref: Some guns/mods with dramatically low AP multis (around 0.7 for 9mm I'm sure you can get even lower) can reduce low AP ammo such as hollowpoint into a negative divisor that shows some funky values such as doing 200-300 **extra** damage to armored targets. These values are only what I see in the info UI they dont seem to be effecting the real bullets. Just weird

Super effectiveness against mobs (and I imagine factions as well but idk) appears totally broken but I'm mostly guessing. Lodge hunting arrows was 2-3 several fully charged shots to kill a common spider. It should be doing over 300 damage from memory with hunting broadheads. Bone rifle was highly ineffective (compared to before) on Superior mobs but plenty effective on simple (due to their math being broken I assume bone rifle is now one shot for a render hehehee).

Lodges hunting lance seems...really depressing frankly. Taking several shots (used to be 2 AP bolts on avg) to kill a common voidwolf its pretty much useless. This used to be comparable to a reductor! Most of its damage came from AP I believe so it needs a looking at.

**Comment by benj8560** [comment]
Melee weapons are not working as expected in PVP. Attacked Trilby in riot armor with a katana has 33 damage and 1.4 AP and 45 rob. 
Did 30 damage??

Figured maybe rob AP scaling was doing it. I tried at 0 robustness got 22 damage. 25 at 10 rob

**Comment by cdb-is-not-good** [comment]
AFAIK melee is mostly not factored in here because of...reasons, IIRC eris was waiting for hteir full on melee rework but melee working a bit differently to range is...fiiiiiiine for now. I'll see what i can do about fixiing supereffective mult

**Comment by Legate-Maxson** [comment]
[https://cdn.discordapp.com/attachments/385878837004861454/1207126185620803614/image.png?ex=65de82dd&is=65cc0ddd&hm=1baa97f08c3dd9c4b9fa5cee0f457d3be99fd212e43f921883c4dc197fdaba2a&](url)

Foreman's armor is paper thin, Worse then most shirts. The helmet has the correct stats.

[https://cdn.discordapp.com/attachments/859890910027841618/1207123862903324702/image.png?ex=65de80b4&is=65cc0bb4&hm=ca1f2766fd14dfd5693d9547cd43dc5d0b2290b7bd6bef6692349f79890c7d00&](url)

You can one shot ravagers with the embedded luty right now.

**Comment by cdb-is-not-good** [comment]
> armor
> 
> Foreman's armor is paper thin, Worse then most shirts. The helmet has the correct stats.
> 
> luty
> 
> You can one shot ravagers with the embedded luty right now.

The numbers listed in the info tab don't seem genuinely representitive but I'll figure out how to make the math not suck, also re; foremans armor 

image

**Comment by cdb-is-not-good** [comment]
Mkay this is being worked on again but is ultimately still not ready to ship. currently got some runtimes and probably a heap of bugs.

File: code/modules/clothing/suits/armor.dm [hunk 6]
```diff
@@ -79,7 +79,7 @@
 	desc = "An armored vest of dubious quality. Provides decent protection against physical damage, for a piece of crap."
 	icon_state = "armor_handmade"
 	armor_list = list(
-		melee = 30,
+		melee =7,
 		bullet = 20,
 		energy = 15,
```
Comment (SingingSpock) [inline code/modules/clothing/suits/armor.dm:84]: Bullet and Energy armor need to be adjusted down to make this not god armor

Comment (cdb-is-not-good) [reply code/modules/clothing/suits/armor.dm:84]: ah piss

**Comment by Dimasw99** [comment]
A lot of armors had its bomb protection changed as well. Was this intentional?

Also the UI is a bit misleading
grafik

**Comment by Dimasw99** [comment]
After playing a round with this yeah. A lot of creatures are now overtuned in their HP pool being roughly twice as strong or even larger depending on what weapon you use. Combine that with the several nerfs to ammo storage and capacity it makes fighting mobs a real chore and constant resupply run. It took me roughly 3 to 4 shots to kill a guardian spider with a shotgun using slugs which otherwise would have been either a oneshot or two shot.

I fear we might need the dreaded PVE damage back.

**Comment by cdb-is-not-good** [comment]
Something that's on the itinerary is a set of defines that will be given to all mobs(by subtype, e.g roaches, spiders, greyson, void wolf, psion) that adjusts their total health by a value that equals one by default. This way we can more easily tweak this moving forward.

**Comment by hex37** [comment]
IMO, the swing-windup-based-on-size thing needs to be removed. We have way too many things that are just arbitrarily big so players don't try to equip them. This kinda flies in the face of that functionality if there's nothing to replace it.

**Comment by cdb-is-not-good** [comment]
ehhh, there's an easy enough fix. A single variable can be slapped on whatever arbitrarily large items we find and so far i've found several and done so.  That functionality was also and is still kinda crap and if we want players not bagging stuff there are other, better ways to do it.

**Comment by Dimasw99** [comment]
rip 
grafik

File: interface/skin.dmf [hunk 7]
```diff
@@ -581,16 +581,7 @@ macro "hotkeymode"
...
-	elem
-		name = "CTRL+F"
-		command = "a-intent left"
-	elem
-		name = "G"
-		command = "a-intent right"
-	elem
-		name = "CTRL+G"
-		command = "a-intent right"
+		command = "blocking"
```
Comment (hex37) [inline interface/skin.dmf:584]: Here is where they ruined everything

Probably

File: interface/skin.dmf [hunk 8]
```diff
@@ -580,17 +580,8 @@ macro "hotkeymode"
 		name = "CTRL+E"
 		command = "quick-equip"
 	elem
-		name = "F"
```
Comment (hex37) [inline interface/skin.dmf:583]: You still need to add back the old hotkeys to enable the cycling again.

Comment (cdb-is-not-good) [reply interface/skin.dmf:583]: Huh, are the cycling hotkeys not existent further down the list? It looked like they were still there

**Comment by cdb-is-not-good** [comment]
lol oops ignore the name/desc of that last commit. leftover commit mess from when i was fucking with maybe merging damtypes 3 into this PR. it's actually just me cherry picking a PR into mine to fix conflicts B)

**Comment by cdb-is-not-good** [comment]
lol doesnt compile cool

**Comment by cdb-is-not-good** [comment]
By the will of the people(and so coders have a less annoying time working around conflicts) this is being merged in what i consider a semi-acceptable state. Further bugfixes/balance changes will come down the line.

**Pull request merged**